### PR TITLE
Fix Issue 21204 - Error in generated copy constructor gives confusing message

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: src/dmd/declaration.d [hunk 1]
```diff
@@ -362,6 +362,14 @@ extern (C++) abstract class Declaration : Dsymbol
                                     return false;
                         }
                     }
+                    if (auto ctor = isCtorDeclaration())
+                    {
+                        if (ctor.isCpCtor && ctor.generated)
+                        {
+                            .error(loc, "Generating an `inout` copy constructor for `struct %s` failed, therefore instances of it are uncopyable", parent.toPrettyChars());
+                            return true;
```
Comment (thewilsonator) [inline src/dmd/declaration.d:370]: This early return is good but the whole function should use it.

Comment (RazvanN7) [reply src/dmd/declaration.d:370]: I know, but I didn;t want to mix refactorings in this PR.

**Comment by RazvanN7** [comment]
I think that this is good to go.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Please do do the refactoring.

**Pull request merged**